rfc/0009-expressions/README.md

```diff
@@ -0,0 +1,94 @@
+| Status | `DRAFT` | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `georg-schwarz` | <!-- TODO: assign yourself as main driver of this RFC -->
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+This RFC introduces a simple expression language for numeric and boolean values.
+
+## Motivation
+
+Right now, users have to to define property values of fractions in the decimal representation, e.g., `0.3`. They cannot express non--terminating decimal values or repeating decimal ones, e.g., `1/3` (a third). 
+
+For future features (like value transformations), we also need boolean expressions that allow for numeric comparisons, e.g. `3 > 5` should resolve to `false`.
+
+## Explanation
+
+### Operators
+
+The following operators are in descending order by their precedence:
+- `()` for grouping a sub-expression
+- unary operators (see below)
+- binary operators (see below)
+
+#### Unary Prefix Operatory
+Binary prefix operators follow the pattern `op a`. Brackets can be used for grouping larger sub-expressions.
+- `floor`, `ceil`, `round` for conversion of decimal numbers to integers.
+- `not` for unary inversion of boolean values
+
+#### Binary Operators
+Binary infix operators follow the infix pattern `a op b`.
+- `pow`, `root` for power and root calculation. E.g., `2 pow 3` is 2³, `5 root 2` is `sqrt(5).
+- `*`, `/`, `%` for multiplication, division, and remainder
+- `+`, `-` for addition and subtraction
+- `<`, `<=`, `>`, `>=` for relational operators comparing numbers
+- `==`, `!=` for equality operators comparing numbers and booleans
+- `and` for a logical AND on booleans
+- `xor` for a logical XOR on booleans
+- `or` for a logical OR on booleans
+
+
+### Handling of booleans
+
+The expression resolving of boolean does follow the common pattern. 
+When operators of similar precedence don't determine the order, we evaluate from left to right.
+
+
+### Handling of numbers
+
+Division and remainder operators **always** produce `decimal` values. To convert to an integer value, the operators `floor`, `ceil`, `round` have to be used.
+
+Division by zero should produce an `INVALID` value in the future. For now, it will just throw an error. 
+
+### Interplay of booleans and numbers
+
+We infer the datatype based on the operator. The resulting value type of each operator unambiguous. 
+
+The operands `==` and `!=` are the only ones that allow operands of different types. Other operators are unambiguous. Comparing the equality of two operands of different value types always resolves to `false`.  
```

Comment:
How about `3/3 == floor (3/3)`? For the record I am fine with:

- Comparing equality of two operands of different value types to always throw an error
- Comparing equality of two operands of different value types to always resolve to `false` and raise a warning

Reply:
Makes sense to me. Any objections @rhazn ?

Reply:
Making it work I think is good. Calling it sub-type is not very good if it is not practically also a sub-type (e.g. if our built-in valuetype integer is not a sub-type of decimal). But in general, I am okay with error but allowing comparison of decimals and integers with a warning.

Reply:
Phrased it like this: [0fa04db](https://github.com/jvalue/jayvee/pull/273/commits/0fa04db7edd618099aa046eb10f6649d60306a18)